gene/etl/ncbi.py

```diff
@@ -237,6 +230,215 @@ def _load_data(self, gene: Gene, batch):
+
+            arm_match = re.search("[pq]", loc)
+            if arm_match and not contains_centromere:
+                arm_ix = arm_match.start()
+                chromosome = loc[:arm_ix].strip()
+
+                # NCBI sometimes stores invalid map locations
+                # i.e. 7637 stores 'map from Rosati ref via FISH [AFS]'
+                if not re.match("^([1-9][0-9]?|X|Y|MT)$", chromosome):
+                    continue
+                location['chr'] = chromosome
+
+                # Check to see if there is a band / sub band included
+                if arm_ix != len(loc) - 1:
+                    if '-' in loc:
+                        self._set_interval_range(loc, arm_ix, interval)
+                    else:
+                        # Location only gives start
+                        start = loc[arm_ix:]
+                        interval['start'] = start
+                        interval['end'] = start
+                else:
+                    # Only arm is included
+                    interval['start'] = loc[arm_ix]
+                    interval['end'] = loc[arm_ix]
+            elif contains_centromere:
+                self._set_centromere_location(loc, location, interval)
+            else:
+                # Location only gives chr
+                location = None
+                interval = None
+                if 'chr' in params['location_annotations']:
+                    params['location_annotations']['chr'].append(loc)
+                else:
+                    params['location_annotations']['chr'] = [loc]
+
+            if location and interval:
+                interval['type'] = IntervalType.CYTOBAND.value
+                location['interval'] = interval
+                location['species_id'] = 'taxonomy:9606'
+                location['type'] = LocationType.CHROMOSOME.value
+                assert ChromosomeLocation(**location)
+                location_list.append(location)
+
+    def _set_interval_range(self, loc, arm_ix, interval):
+        """Set the location interval range.
+
+        :param str loc: A gene location
+        :param int arm_ix: The index of the q or p arm for a given location
+        :param dict interval: The GA4GH interval for a VRS object
+        """
+        # Location gives both start and end
+        range_ix = re.search('-', loc).start()
+
+        start = loc[arm_ix:range_ix]
+        start_arm_ix = re.search("[pq]", start).start()
+        start_arm = start[start_arm_ix]
+
+        end = loc[range_ix + 1:]
+        end_arm_match = re.search("[pq]", end)
+
+        if not end_arm_match:
+            # Does not specify the arm, so use the same as start's
+            interval['start'] = start
+            interval['end'] = f"{start[0]}{end}"
+        else:
+            end_arm_ix = end_arm_match.start()
+            end_arm = end[end_arm_ix]
+
+            # GA4GH: If start and end are on the same arm,
+            # start MUST be the more centromeric position
```

Comment:
It might not matter since it may end up changing, but I think it can be helpful to link to sources for stuff like this when you're explaining decisions based on specs or standards for people who aren't familiar with it

Reply:
Added 👍